### PR TITLE
🎨 Reorganise units example to ParameterFrame example

### LINKED ISSUE #1426: Write documentation for new parameter classes
## Description

The new `Parameter` and `ParameterFrame` classes need rst documentation and code examples.

## Outputs

- Jupyter notebook example for usage of `Parameter`
- Jupyter notebook example for usage of `ParameterFrame`
- RST documentation for Bluemira paramters and paramter frame.

### PR DESCRIPTION
**Pull request by je-cook**
## Linked Issues

Closes #1426

## Description

There was no explicit `ParameterFrame` example. This adjusts the units example to fill that gap

## Interface Changes

None

## Checklist

I confirm that I have completed the following checks:

- [X] Tests run locally and pass `pytest tests --reactor`
- [X] Code quality checks run locally and pass `flake8` and `black .`
- [X] Documentation built locally and checked `sphinx-build -W documentation/source documentation/build`

### CODE REVIEW

File: examples/base/params.py [hunk 1]
```diff
@@ -37,35 +41,9 @@
...
-    const.to_kelvin(-300)
-except ValueError as v:
-    print(v)
-
-print(const.to_celsius(10, unit="rankine"))
-
 # %%[markdown]
 # ## Parameters and Units
-# First I make a small ParameterFrame
+# First I make a small ParameterFrame and compare the two methods of creating a frame
```
Comment (hsaunders1904) [inline examples/base/params.py:46]:
```suggestion
# First, I make a small `ParameterFrame` and compare the two methods of creating it.
```

File: examples/base/params.py [hunk 2]
```diff
@@ -82,11 +61,16 @@ class MyParameterFrame(ParameterFrame):
...
 
+    R_0: Parameter[float]
     A: Parameter[float]
 
 
-mypf = MyParameterFrame.from_dict({"A": {"value": 5, "unit": ""}})
-mydecpf = MyDecoratedFrame.from_dict({"A": {"value": 5, "unit": ""}})
+mypf = MyParameterFrame.from_dict(
+    {"A": {"value": 5, "unit": ""}, "R_0": {"value": 8, "unit": "m"}}
+)
```
Comment (hsaunders1904) [inline examples/base/params.py:70]: A comment here noting that an empty unit is equivalent to `"dimensionless"` may be helpful.

File: examples/base/params.py [hunk 3]
```diff
@@ -109,6 +93,46 @@ class MyDecoratedFrame:
...
+# %%
+
+print(mypf.R_0.value_as("cm"))  # 600
+
+# %%[markdown]
+# ## Raw conversion
+
+# In some situations it may be useful to convert a raw value to a different unit.
+# It is recommended (and enforced within bluemira core) that all conversions however
+# simplistic are converted this way.
```
Comment (hsaunders1904) [inline examples/base/params.py:113]: 'enforced' feels like the wrong word here. If I'm understanding you correctly, maybe something like this:

```suggestion
# It is recommended that all conversions, 
# however simple, are made using `bluemira.constants.raw_uc`.
# This is how all unit conversions are performed internally.
```

File: examples/base/params.py [hunk 4]
```diff
@@ -109,6 +93,46 @@ class MyDecoratedFrame:
...
+#
+# Any possible future base unit changes can be achieved by searching for a given unit.
```
Comment (hsaunders1904) [inline examples/base/params.py:115]: I'm not entirely sure what you mean here.

**Review by hsaunders1904 — CHANGES_REQUESTED** [review]
It feels a little strange to me that we've got documentation for `raw_uc` within the example for parameters. Maybe it's worth us writing some RST docs specifically for units and unit conversions? We can put some very basic examples in there. I'm happy to open an issue & write that up.

Also, I'm actually getting an error running this example:

```
Traceback (most recent call last):
  File "/home/bf2936/bluemira/code/bluemira/bluemira/base/parameter_frame/_frame.py", line 118, in from_dict
    param_data = data.pop(member)
KeyError: 'R_0'

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/home/bf2936/bluemira/code/bluemira/examples/base/params.py", line 84, in <module>
    mydiffval = MyDecoratedFrame.from_dict({"A": {"value": 6, "unit": "m"}})
  File "/home/bf2936/bluemira/code/bluemira/bluemira/base/parameter_frame/_frame.py", line 120, in from_dict
    raise ValueError(f"Data for parameter '{member}' not found.") from e
ValueError: Data for parameter 'R_0' not found.
```

File: examples/base/params.py (hunk 4 above)
Comment (je-cook) [reply examples/base/params.py]: Reworded in units.py

File: examples/base/params.py [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+# # Parameters and ParameterFrames in bluemira
+# `Parameters` and `ParameterFrames` are the mechanism bluemira uses to contain
+# metadata about a given value.
+# Each `Parameter` must have a unit associated with the value and can have a
+# source, description and long_name.
+#
+# The mechanics of the unit system in bluemira are fairly staight forward
+# It provides an implicit interface to convert units to the
+# [base units of bluemira](
+# https://bluemira.readthedocs.io/en/latest/conventions.html#unit-convention).
```
Comment (hsaunders1904) [inline examples/base/params.py:36]: I'm not sure what you mean here. Maybe something like this?

> \`ParameterFrame\`s implicitly convert parameter values' units to the \[base units of bluemira\](
https://bluemira.readthedocs.io/en/latest/conventions.html#unit-convention).

Comment (je-cook) [reply examples/base/params.py]: Sure :)

**Review by hsaunders1904 — APPROVED** [review]

**Pull request merged**